### PR TITLE
Update resource manager to return TaskResourceAttributes

### PR DESCRIPTION
**Pull request by wild-endeavor**
# TL;DR
Add task resources to the list of matchable attributes that we fall back to system settings for.

Changed the resource manager to take in the very top level `runtimeInterfaces.Configuration` interface instead of just `ApplicationConfiguration`.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Added `GetAsWorkflowExecutionAttribute()` and `GetAsAttribute()` to the `ApplicationConfiguration` and  `TaskResourceConfiguration` interfaces respectively.

## Tracking Issue
https://github.com/flyteorg/flyte/issues/3467

### CODE REVIEW

File: pkg/manager/impl/util/shared.go [hunk 1]
```diff
@@ -327,3 +328,26 @@ func MergeIntoExecConfig(workflowExecConfig admin.WorkflowExecutionConfig, spec
...
+	if (higher.GetGpu() == "" || higher.GetGpu() == "0") && len(lower.GetGpu()) > 0 {
+		higher.Gpu = lower.GetGpu()
+	}
+	if (higher.GetMemory() == "" || higher.GetMemory() == "0") && len(lower.GetMemory()) > 0 {
+		higher.Memory = lower.GetMemory()
+	}
+	if (higher.GetEphemeralStorage() == "" || higher.GetEphemeralStorage() == "0") && len(lower.GetStorage()) > 0 {
+		higher.Storage = lower.GetStorage()
+	}
+	return &higher
```
Comment (pingsutw) [inline pkg/manager/impl/util/shared.go:345]: should we add `EphemeralStorage` here

**Review by pingsutw — COMMENTED** [review]
LGTM
